src/txn/txn_rollback_to_stable.c

```diff
@@ -234,8 +234,8 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
          * Verify the history store timestamps are in order. The start timestamp may be equal to the
          * stop timestamp if the original update's commit timestamp is out of order.
          */
-        WT_ASSERT(session,
-          (newer_hs_ts == WT_TS_NONE || hs_stop_ts <= newer_hs_ts || hs_start_ts == hs_stop_ts));
+        WT_ASSERT(session, (newer_hs_ts == WT_TS_NONE || hs_stop_ts <= newer_hs_ts ||
```

Comment:
Does this out of order issue cause other visibility problems in searching?

Reply:
Does `hs_start_ts == newer_hs_ts` cover all the out of order cases? What about in the original update chain, we have:
U@20 -> U@10 -> T@15 -> U@5
I think the assert will still fire.

Reply:
Should be `hs_start_ts <= newer_hs_ts`